src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -52,7 +51,29 @@ trait StandardScalaSettings { _: MutableSettings =>
   val nowarn =         BooleanSetting ("-nowarn", "Generate no warnings.") withAbbreviation "--no-warnings" withPostSetHook { s => if (s) maxwarns.value = 0 }
   val optimise:        BooleanSetting // depends on post hook which mutates other settings
   val print =          BooleanSetting ("-print", "Print program with Scala-specific features removed.") withAbbreviation "--print"
-  val target =         ChoiceSetting  ("-target", "target", "Target platform for object files.", AllTargetVersions, "8") withPreSetHook normalizeTarget _ withAbbreviation "--target"
+  val release =
+    ChoiceSetting("-release", "release", "Compile for a version of the Java API and target class file.", AllTargetVersions, normalizeTarget(javaSpecVersion))
+      .withPostSetHook { setting =>
+        val current = setting.value.toInt
+        if (!isJavaAtLeast("9") && current > 8) errorFn.apply("-release is only supported on JVM 9 and higher")
+        if (target.valueSetByUser.map(_.toInt > current).getOrElse(false)) errorFn("-release cannot be less than -target")
+        //target.value = setting.value  // this would trigger deprecation
+      }
+      .withAbbreviation("--release")
+      // .withAbbreviation("-java-output-version")
```

Comment:
In Scala 3, [`-java-output-version` is the preferred flag name](https://github.com/lampepfl/dotty/pull/14606) (with the alias of `-release` still permitted) - would it be reasonable to uncomment this line, and allow Scala 2.13 to use `-java-output-version` as well?

Reply:
I believe we'd accept a PR that did that — at least I can't think why not

Reply:
Because the dotty project is really bad at choosing option names.

Reply:
(to be clear: not suggesting we make it canonical in Scala 2, just an accepted alternative)

Reply:
see also `-Xunchecked-java-output-version`. I didn't want to encourage them (the folks inventing these names). I came close to enabling them (the alt names) on this PR.

Reply:
> I believe we'd accept a PR that did that — at least I can't think why not

Thanks! Opened with https://github.com/scala/scala/pull/10654.

> Because the dotty project is really bad at choosing option names.

This comment reminded me of my favourite video on software development, the 2012 talk by Andrew Dupont [_How to Argue About Code_](https://vimeo.com/53218578#t=477s) (sadly, as it's 'Not Yet Rated' on Vimeo, you have to login to view it in UK & EU, but at least after that there's no ads).

Regarding the option name, for myself, when I'm trying to help junior devs understand that if they want to run their code on Java 11, they need to specify `-java-output-version 11`, the conversation is a little easier than it is if I say `-release 11`. The first name, because it includes 'java', is more self-explanatory. The junior dev is _aware_ of Scala and Java, but doesn't automatically realise that when Scala says `-release`, it's referring to a Java-related concern.

Reply:
I'm revisiting the PR over why target was deprecated, and wanted to clarify that I was teasing about option names. This is all in the service of making it easy for people to compile stuff safely.